### PR TITLE
Tmz 1.6.0 shadowlength

### PR DESCRIPTION
**Pull request by t-m-z**
something to discuss
the tool is an extensive reuse od the sunposition widget.
however, I compared the results with different websites 
  https://de.planetcalc.com/318/
  http://www.volker-lotze.de/content/schattenrechner/schattenrechner.php
  https://rechneronline.de/sehwinkel/schattenlaenge.php
  https://www.sonnenverlauf.de/
  Helios 2.2 - software to calculate shadows
and got different results. I trust in your algorithms but I am

### CODE REVIEW

**Comment by S-Man42** [comment]
please resolve the conflicts here

File: assets/i18n/en.json [hunk 1]
```diff
@@ -2379,6 +2379,12 @@
   "settings_coordinates_defaultrotationellipsoid_user": "Specify own",
   "settings_mayacalendar_correlation_number_title" :  "Settings: Maya Calendar",
   "settings_mayacalendar_correlation_number" :  "Correlation number",
+  "shadowlength_title": "Shadow length",
```
Comment (S-Man42) [inline assets/i18n/en.json:2382]: Please capital letter always ;)

**Comment by S-Man42** [comment]
For example: Here I miss the entry into the main_view ;)

**Comment by S-Man42** [comment]
And I miss the logic... Am I right?

**Comment by S-Man42** [comment]
Discussion: If you found better and more precise - free usable - algorithms for the sun data, please feel free to update the code! :)

**Comment by t-m-z** [comment]
there is no need for a logic file
it is only a trigonometric calculation with the tangens of the suns elevation

Sent from MailDroid

-----Original Message-----

File: assets/i18n/en.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/en.json:2382]: done

**Comment by t-m-z** [comment]
> For example: Here I miss the entry into the main_view ;)

Hm - Ok - so I still have a lack of understanding. Up to now I thought only the widgets which are directly accessible have to be an entry in the mainview.
That means - every tool has to be an entry in the main_view?

**Comment by t-m-z** [comment]
> Discussion: If you found better and more precise - free usable - algorithms for the sun data, please feel free to update the code! :)

Ok - I will ask the website-owners, which algorithms they used and then do a comparison.

Which is your reference/source?

**Comment by S-Man42** [comment]
You can see it in the astronomy logic directory. Its either jgiesen.de or astronomie.info

**Comment by S-Man42** [comment]
What's the progress here? (Mind the conflict...)

**Comment by t-m-z** [comment]
Still trying to read and undersatand the different algorithms – but due tot he missing air condition in my office under the roof I need some more time

 

Von: MLorenz42 ***@***.***> 
Gesendet: Monday, 28 June 2021 11:59
An: S-Man42/GCWizard ***@***.***>
Cc: Thomas ***@***.***>; Author ***@***.***>
Betreff: Re: [S-Man42/GCWizard] Tmz 1.6.0 shadowlength (#565)

 

What's the progress here? (Mind the conflict...)

**Comment by S-Man42** [comment]
Any progress here? Pospone to later version (close PR)?

**Comment by t-m-z** [comment]
Assuming your algorithm to calculate the sun position is correct, the feature can be merged.
however - I still have no clue where the differences in calculating the sun position come from comparing to Volker Lotze and other websites.

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
The en.lang (title, description) is missing here.

**Comment by t-m-z** [comment]
> The en.lang (title, description) is missing here.

were added again, was probably a merge issue

**Pull request merged**